src/player.cpp

```diff
@@ -181,12 +193,346 @@ namespace openre::player
+            if (key_trg & 0x0200)
+            {
+                // trigger quickturn
+                *(uint32_t*)&player->routine_0 = 0xC01;
+                return;
+            }
+            if ((key_trg & 0x80) != 0)
+            {
+                player->status_flg |= 0x200000;
+            }
+            if (key & 0x100 && player->type & 0xFFF)
+            {
+                *(uint32_t*)player->routine_0 = 0x501;
+            }
+        }
+        else
+        {
+            *(uint32_t*)&player->routine_0 = 1;
+
+            if (key & 0xA)
+            {
+                *(uint32_t*)&player->routine_0 = 0x401;
+            }
+
+            if (key & 1)
+            {
+                *(uint32_t*)&player->routine_0 = 0x101;
+            }
+        }
+    }
+
+    // no input code is required for quickturn
+    void pl_br_quickturn(PlayerEntity* player, uint32_t key, uint32_t key_trg) {}
+
+    void pl_mv_quickturn(PlayerEntity* player, uint32_t pKanPtr, uint32_t pSeqPtr)
+    {
+        switch (player->routine_2)
+        {
+        case 0:
+            player->routine_2 = 1;
+            player->timer0 = 0;
+            // set base walk animation
+            *(uint32_t*)&player->move_no = 0x070000;
+            [[fallthrough]];
+
+        case 1:
+            if (player->timer0++ < 8)
+            {
+                player->cdir.y += 2048 / 8;
+            }
+            else
+            {
+                *(uint32_t*)&player->routine_0 = 1;
+            }
+            break;
+        }
+
+        if (*player->pNow_seq & 0x4000)
+        {
+            snd_se_walk(0, 3 * ((*player->pNow_seq >> 13) & 1) + 4, player);
+            gGameTable.word_989EEE |= 2;
+        }
+        joint_move(player, player->pKan_t_ptr, player->pSeq_t_ptr, 512);
+    }
+
+    void init_quickturn_move()
+    {
+        // fill expanded tables with old code
+        std::memcpy(br_tbl, gMoveBrTable, 12 * 4);
+        std::memcpy(mv_tbl, gMoveMvTable, 12 * 4);
+        // set hooks for quickturn
+        br_tbl[03] = pl_br_03;
+        br_tbl[12] = pl_br_quickturn;
+        mv_tbl[12] = pl_mv_quickturn;
+        // replace old table pointers
+        gMoveBrTable = br_tbl;
+        gMoveMvTable = mv_tbl;
+    }
+
     void player_init_hooks()
     {
         interop::writeJmp(0x00502190, &partner_switch);
         interop::writeJmp(0x00502660, &inventory_find_item);
         interop::writeJmp(0x4FC3CE, itembox_prev_slot);
         interop::writeJmp(0x5024D0, set_inventory_item);
         interop::writeJmp(0x502500, set_inventory_item_quantity);
+        interop::writeJmp(0x4D97B0, player_move);
+        interop::writeJmp(0x4D9D20, pl_move);
+        interop::writeJmp(0x4DAE70, player_mv_rotate);
+        interop::writeJmp(0x004DA6C0, pl_br_03);
+        init_quickturn_move();
+    }
+
+    bool is_aiming()
+    {
+        return (
+            check_flag(FlagGroup::Status, FG_STATUS_PLAYER) && check_flag(FlagGroup::Status, FG_STATUS_GAMEPLAY)
```

Comment:
How come you are checking player flag? That just says whether you are leon or claire.

Reply:
Good point, I don't remember why I did that.

After removing that flag I see some errors on `is_picking_up_item` function, as is not working yet and we do not needed for the moment I removed it. `is_aiming` seems to be correct.